### PR TITLE
core: More fixes for mixed AVM

### LINKED ISSUE #10759: Error on Jump!
# Error Info
Error name: Error
Error message: panicked at 'No script object for display object', core/src/avm1/activation.rs:396:18
Error stack:
```
Error: panicked at 'No script object for display object', core/src/avm1/activation.rs:396:18
    at O.n.wbg.__wbg_new_15d3966e9981a196 (https://www.spiele-umsonst.de/ruffle/core.ruffle.a648731f80678e415588.js:1:47949)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::hbac0af9b7f44ed2e (wasm://wasm/02a7785a:wasm-function[8134]:0x7ee7f9)
    at std::panicking::rust_panic_with_hook::hc53aea0352e77326 (wasm://wasm/02a7785a:wasm-function[5462]:0x727498)
    at core::panicking::panic_fmt::hf4a9df75710ece83 (wasm://wasm/02a7785a:wasm-function[8016]:0x7e9468)
    at ruffle_core::display_object::edit_text::EditText::try_bind_text_field_variable::h5e7c07cdf91e2471 (wasm://wasm/02a7785a:wasm-function[864]:0x2d1bb9)
    at <ruffle_core::display_object::DisplayObject as ruffle_core::display_object::TDisplayObject>::post_instantiation::h1dba24217afabf1d (wasm://wasm/02a7785a:wasm-function[2847]:0x5c2184)
    at ruffle_core::display_object::movie_clip::MovieClip::instantiate_child::hd84fb6f602ec2e60 (wasm://wasm/02a7785a:wasm-function[777]:0x293b4b)
    at ruffle_core::display_object::movie_clip::MovieClip::place_object::h74a63696e0e4a88f (wasm://wasm/02a7785a:wasm-function[1563]:0x41dbe8)
    at <ruffle_core::display_object::movie_clip::MovieClip as ruffle_core::display_object::TDisplayObject>::enter_frame::h3c3024109b2e33ab (wasm://wasm/02a7785a:wasm-function[788]:0x29abfb)
    at <ruffle_core::display_object::loader_display::LoaderDisplay as ruffle_core::display_object::TDisplayObject>::enter_frame::hcf918fd1eadcdac3 (wasm://wasm/02a7785a:wasm-function[3019]:0x5ee6d2)
```

# Player Info
Allows script access: false

# Page Info
SWF URL: https://www.spiele-umsonst.de/azad/downloads/flashgames/doodlejump.swf

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-04-19
Channel: nightly
Built: 2023-04-19T00:19:17.744Z
Commit: 4f4327b152087379bae0d566a5d80aa7c9b9c477
Is extension: false

# Metadata
width: 400
height: 600
frameRate: 30
numFrames: 2
swfVersion: 10
backgroundColor: #FFFFFF
isActionScript3: true
uncompressedLength: 636899

### LINKED ISSUE #14264: Bloons: Doppleprimeape issue
### Describe the bug

Game used to have an issue where it zoomed through all the screens endlessly but its fixed now, instead sometimes when you go to the next level, the game will spawn a dopplemonkey next to your monkey that never goes away. Most consistent way to trip it is to beat level one, then click next level and it'll happen on level 2.

Screenshot_20231130_184814_Chrome

### Expected behavior

There shouldn't be any duplicated monkeys next to each other
Screenshot_20231130_184754_Chrome

### Content Location

[bloons.zip](https://github.com/ruffle-rs/ruffle/files/13519662/bloons.zip)

### Affected platform

Online demo

### Operating system

Android 13

### Browser

Chrome 119

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Lord-McSweeney**
Fixes #14264, closes #10759

### CODE REVIEW

**Comment by Lord-McSweeney** [comment]
I need a little more time to figure out some of the correct behaviors

**Comment by Lord-McSweeney** [comment]
This relies on #13889

**Comment by torokati44** [comment]
> This relies on https://github.com/ruffle-rs/ruffle/pull/13889

But that is still a draft. Is it because of @Toad06's concern?

File: core/src/loader.rs [hunk 1]
```diff
@@ -42,6 +42,8 @@ use url::{form_urlencoded, ParseError, Url};
 
 pub type Handle = Index;
 
+const LOADER_INSERTED_AVM1_DEPTH: i32 = -0xF000;
```
Comment (torokati44) [inline core/src/loader.rs:45]: A short, one-line comment on what this is would be nice. It's almost self-explanatory, but still.

File: tests/tests/swfs/avm2/mixed_avm/output.txt [hunk 2]
```diff
@@ -1,5 +1,11 @@
 Starting load...
+_global access in DoInitAction: 
+[type Object]
+root depth:
+-77824
 hello from avm1!
+Root depth: -77824
+_root == _level0: false
```
[the remark this answers is not in the document]
Comment (Lord-McSweeney) [reply tests/tests/swfs/avm2/mixed_avm/output.txt:8]: What's the issue?

[the remark this answers is not in the document]
Comment (torokati44) [reply tests/tests/swfs/avm2/mixed_avm/output.txt:8]: No issue whatsoever! It's just a comment of approval!

**Review by Dinnerbone — APPROVED** [review]

File: core/src/loader.rs (hunk 1 above)
Comment (torokati44) [reply core/src/loader.rs:45]: Would you mind if I added this myself before merging, @Lord-McSweeney?
`/// This is the depth where AVM2 puts the AVM1 movies it loads.`
Is this even correct?

**Comment by torokati44** [comment]
Thanks for the comment, merging!

**Pull request merged**